linux/LibSensors.c

```diff
@@ -261,6 +261,43 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Read sysfs to get HT/SMT cores and correctly distribute the temperature values */
+   if (coreTempCount < activeCPUs) {
+      double oldData[existingCPUs + 1];
+      memcpy(oldData, data, sizeof(oldData));
+
+      int dataCell = 1;
+      for (unsigned int i = 0; i < existingCPUs; i++) {
+         char path[100];
+         sprintf(path, "/sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", i);
+         FILE* file = fopen(path, "r");
+         if (file == NULL) {
+            data[i+1] = NAN;
+            dataCell++;
+            continue;
+         }
+
+         char contents[10];
+         fgets(contents, 10, file);
+         fclose(file);
+
+         int start, end;
+         int rangeNums = sscanf(contents, "%d-%d", &start, &end);
+
+         if (rangeNums == 1) {
+            data[i+1] = oldData[dataCell];
```

Comment:
Inconsistent use of indexing source. When just one value given, this should equal to running the below code for just the start index, which is not the case at all.